consensus/types/src/beacon_state.rs

```diff
@@ -2715,6 +2722,37 @@ impl<E: EthSpec> BeaconState<E> {
 
         Ok(())
     }
+
+    /// Returns the weak subjectivity period for `self`
+    pub fn compute_weak_subjectivity_period(&self, spec: &ChainSpec) -> Result<Epoch, Error> {
+        let total_active_balance = self.get_total_active_balance()?;
```

Comment:
Here we are assuming the total active balance cache is initialized, which is fine (see comment below)